legal-api/migrations/versions/99575010ed4b_create_authorized_role_actions_table.py

```diff
@@ -0,0 +1,361 @@
+"""create_authorized_role_actions_table
+
+Revision ID: 99575010ed4b
+Revises: 1dd76abf128a
+Create Date: 2025-06-04 01:03:37.254302
+
+"""
+from datetime import datetime, timezone
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '99575010ed4b'
+down_revision = '1dd76abf128a'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Create user_roles table
+    op.create_table(
+        'user_roles',
+        sa.Column('id', sa.Integer, primary_key=True),
+        sa.Column('role_name', sa.String(50), nullable=False, unique=True),
+        sa.Column('created_date', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('last_modified', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('created_by_id', sa.Integer, nullable=True),
+        sa.Column('modified_by_id', sa.Integer, nullable=True)
+    )
+
+    # Create actions table
+    op.create_table(
+        'actions',
+        sa.Column('id', sa.Integer, primary_key=True),
+        sa.Column('action_name', sa.String(100), nullable=False, unique=True),
+        sa.Column('description', sa.String(255), nullable=True),
+        sa.Column('created_date', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('last_modified', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('created_by_id', sa.Integer, nullable=True),
+        sa.Column('modified_by_id', sa.Integer, nullable=True),
+    )
+
+    # Create authorized_role_actions table
+    op.create_table(
+        'authorized_role_actions',
+        sa.Column('role_id', sa.Integer, sa.ForeignKey('user_roles.id'), primary_key=True),
+        sa.Column('action_id', sa.Integer, sa.ForeignKey('actions.id'), primary_key=True),
+        sa.Column('created_date', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('last_modified', sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column('created_by_id', sa.Integer, nullable=True),
+        sa.Column('modified_by_id', sa.Integer, nullable=True),
+    )
+
+    now = datetime.now(timezone.utc)
+
+    user_roles = sa.table(
+        'user_roles',
+        sa.column('role_name', sa.String),
+        sa.column('created_date', sa.TIMESTAMP(timezone=True)),
+        sa.column('last_modified', sa.TIMESTAMP(timezone=True)),
+        sa.column('created_by_id', sa.Integer),
+        sa.column('modified_by_id', sa.Integer)
+    )
+
+    user_role_data = [
+        {'name': 'contact_centre_staff'},
+        {'name': 'maximus_staff'},
+        {'name': 'public_user'},
+        {'name': 'sbc_staff'},
+        {'name': 'staff'}
```

Comment:
I noticed that some accounts have multiple roles, so there's kind of an order here. Specifically, when checking a token 's roles, "public_user" needs to be the last choice. Is that a problem for the other ticket (endpoint), or do you think this list should be ordered?

Reply:
What is the expectation if an account have 2 roles? allow all actions in both roles or allow actions based on the priority of role?